### PR TITLE
[ADF- 4871][TaskHeaderCloudComponent] Add Candidate Users/Groups properties

### PR DESCRIPTION
**Pull request by sivakumar414ram**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

1. TaskHeaderCloud component does not expose candidate users/groups that were added during modeling time.
2. fullName pipe to just renders first or the last name if they present, if not it just returns an empty string. 

User Header:

image

PeopleCloudSearch component:

image

https://issues.alfresco.com/jira/browse/ADF-4971

**What is the new behaviour?**

1. fullName pipe will render username or email in case first & the last name is missing.

User Header:
image

PeopleCloudSearch component:
image

2.TaskHeaderCloud component displays candidate users/groups that were added during modeling time. To achieve the above behavior we need a new CardView component, which takes an array of values and displays them using mat-chip.

3.New CardViewArrayItemComponent Type created in the card view module:

CardViewArrayItemComponent:
card-view-array-item

card-view-array-with-more-option

card-view-array-with-more-info

TaskHeaderCloudComponent With Candidate Users/Groups:

task-cloud-header-component-1

task-cloud-component-2

With No Candidate User/Groups

task-cloud-header-3

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/card-view/components/card-view-arrayitem/card-view-arrayitem.component.scss [hunk 1]
```diff
@@ -0,0 +1,29 @@
+@mixin adf-card-view-array-item-theme($theme) {
+    $foreground: map-get($theme, foreground);
+    $outline: 1px solid mat-color($alfresco-ecm-blue, A200) !default;
```
Comment (eromano) [inline lib/core/card-view/components/card-view-arrayitem/card-view-arrayitem.component.scss]: this color should be $primary

File: lib/core/card-view/components/card-view-arrayitem/card-view-arrayitem.component.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import { CardViewUpdateService } from '../../services/card-view-update.service';
+
+@Component({
+  selector: 'adf-card-view-arrayitem',
+  templateUrl: './card-view-arrayitem.component.html',
+  styleUrls: ['./card-view-arrayitem.component.scss']
+})
+export class CardViewArrayItemComponent {
+
+    @Input()
```
Comment (eromano) [inline lib/core/card-view/components/card-view-arrayitem/card-view-arrayitem.component.ts]: Those inputs need the jsdoc documentation comment

File: lib/core/card-view/components/card-view-arrayitem/card-view-arrayitem.component.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    clicked(): void {
+        this.cardViewUpdateService.clicked(this.property);
+    }
+
+    hasIcon(): boolean {
+        return !!this.property.icon;
+    }
+
+    displayCount() {
```
Comment (eromano) [inline lib/core/card-view/components/card-view-arrayitem/card-view-arrayitem.component.ts]: missing return type in some method of this class

File: lib/core/card-view/models/card-view-arrayitem.model.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    constructor(cardViewArrayItemProperties: CardViewArrayItemProperties) {
+        super(cardViewArrayItemProperties);
+
+        this.items$ = cardViewArrayItemProperties.items$;
+        this.noOfItemsToDisplay = cardViewArrayItemProperties.noOfItemsToDisplay;
+    }
+
+    get displayValue() {
+            return this.value;
```
Comment (eromano) [inline lib/core/card-view/models/card-view-arrayitem.model.ts]: missing return type and wrong indentation

File: lib/process-services-cloud/src/lib/task/task-header/components/task-header-cloud.component.ts [hunk 5]
```diff
@@ -194,10 +195,40 @@ export class TaskHeaderCloudComponent implements OnInit, OnDestroy {
...
+                    key: 'candidateGroups',
+                    icon: 'person',
+                    default: this.translationService.instant('ADF_CLOUD_TASK_HEADER.PROPERTIES.CANDIDATE_GROUPS_DEFAULT'),
+                    noOfItemsToDisplay: 2
+                }
             )
         ];
     }
 
+    private getCandidateUsers() {
```
Comment (eromano) [inline lib/process-services-cloud/src/lib/task/task-header/components/task-header-cloud.component.ts]: missing return type on both methods

File: lib/core/card-view/components/card-view-arrayitem/card-view-arrayitem.component.scss (hunk 1 above)
Comment (sivakumar414ram) [reply lib/core/card-view/components/card-view-arrayitem/card-view-arrayitem.component.scss]: Removed the CSS, It is not used.

File: lib/core/card-view/components/card-view-arrayitem/card-view-arrayitem.component.ts (hunk 2 above)
Comment (sivakumar414ram) [reply lib/core/card-view/components/card-view-arrayitem/card-view-arrayitem.component.ts]: done

File: lib/core/card-view/models/card-view-arrayitem.model.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+    constructor(cardViewArrayItemProperties: CardViewArrayItemProperties) {
+        super(cardViewArrayItemProperties);
+
+        this.items$ = cardViewArrayItemProperties.items$;
+        this.noOfItemsToDisplay = cardViewArrayItemProperties.noOfItemsToDisplay;
+    }
+
+    get displayValue() {
+        return this.value;
```
Comment (arditdomi) [inline lib/core/card-view/models/card-view-arrayitem.model.ts]: You added all the other return types ( as @eromano commented)  but you forgot this one

File: lib/process-services-cloud/src/lib/task/services/task-cloud.service.spec.ts [hunk 7]
```diff
@@ -312,4 +332,80 @@ describe('Task Cloud Service', () => {
                 done();
             });
     });
+
+    it('should return the candidate users by appName and taskId', (done) => {
+        const appName = 'taskp-app';
+        const taskId = '68d54a8f';
+        spyOn(alfrescoApiMock, 'getInstance').and.callFake(returnFakeCandidateUsersResults);
+        service.getCandidateUsers(appName, taskId).subscribe((res: any) => {
```
Comment (arditdomi) [inline lib/process-services-cloud/src/lib/task/services/task-cloud.service.spec.ts]: Can we do res: string[] ?

File: lib/process-services-cloud/src/lib/task/services/task-cloud.service.spec.ts [hunk 8]
```diff
@@ -312,4 +332,80 @@ describe('Task Cloud Service', () => {
...
+                expect(error).toBe('AppName/TaskId not configured');
+                done();
+            });
+    });
+
+    it('should return the candidate groups by appName and taskId', (done) => {
+        const appName = 'taskp-app';
+        const taskId = '68d54a8f';
+        spyOn(alfrescoApiMock, 'getInstance').and.callFake(returnFakeCandidateGroupResults);
+        service.getCandidateGroups(appName, taskId).subscribe((res: any) => {
```
Comment (arditdomi) [inline lib/process-services-cloud/src/lib/task/services/task-cloud.service.spec.ts]: same here. Can we do res: string[] ?

Comment (sivakumar414ram) [reply lib/process-services-cloud/src/lib/task/services/task-cloud.service.spec.ts]: done

File: lib/core/card-view/models/card-view-arrayitem.model.ts (hunk 6 above)
Comment (sivakumar414ram) [reply lib/core/card-view/models/card-view-arrayitem.model.ts]: done

**Pull request merged**